Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/triangle_soup_snap_rounding.h

```diff
@@ -0,0 +1,585 @@
+// Copyright (c) 2025 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+// Author(s)     : Léo Valque
+
+#ifndef CGAL_POLYGON_MESH_PROCESSING_INTERNAL_POLYGON_SOUP_SNAP_ROUNDING_H
+#define CGAL_POLYGON_MESH_PROCESSING_INTERNAL_POLYGON_SOUP_SNAP_ROUNDING_H
+
+#include <CGAL/license/Polygon_mesh_processing/geometric_repair.h>
+
+#include <CGAL/Algebraic_structure_traits.h>
+#include <CGAL/number_utils.h>
+
+#include <CGAL/Polygon_mesh_processing/repair_polygon_soup.h>
+#include <CGAL/boost/graph/named_params_helper.h>
+
+#include <CGAL/Fraction_traits.h>
+#include <CGAL/Lazy_exact_nt.h>
+
+namespace CGAL
+{
+
+namespace Polygon_mesh_processing
+{
+
+namespace autorefine_impl
+{
+
+// Certified ceil function for exact number types
```

Comment:
@sloriot is that something of general interest and should go in number_utils or similar?